Content.Server/Weapon/Ranged/ServerRangedWeaponComponent.cs

```diff
@@ -159,6 +159,12 @@ private void TryFire(EntityUid user, Vector2 targetPos)
                 return;
             }
 
+            _entMan.TryGetComponent(user, out TransformComponent transform);
+            if ((transform.WorldPosition - targetPos).Length  < 1.5)
+            {
+                return;
+            }
```

Comment:
This should check for melee combat being handled as not being able to shoot something on you is a bit sus.

Reply:
I'm not sure how I would check that, because I don't know how I would subscribe to an event to check that it's handled, since the current code is in a component.

Reply:
h m m you could make left-click always shoot with a gun and make the alternate interaction key (I think it's e by default) do melee where something has both gun and meleeweaponcomponent?

Reply:
> This should check for melee combat being handled as not being able to shoot something on you is a bit sus.

Now that guns have been ECS'd how would I access two events in one handler, so I could "listen" for a melee event and check that it's handled before cancelling the shot.

Or maybe there's a better way to do this and I'm just dumb.

Reply:
@metalgearsloth is there a way to do this^
Or how should I do it then.

Reply:
So ideally you'd just have the alt interaction just do melee combat (it probably needs a new keybind). Gun code itself is always just going to receive the shooting events from the client so this would be done separately.